### ISSUE TITLE
Wrong error message for testing `void` with `??`

### ISSUE THREAD

**Issue by nicolo-ribaudo** [microsoft/TypeScript#57172]
### 🔎 Search Terms

nullish An expression of type 'void' cannot be tested for truthiness.

### 🕗 Version & Regression Information

- This changed in commit or PR _______
- This is the behavior in every version I tried

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=5.3.3#code/CYUwxgNghgTiAEA3W8oC4kHsCWwDcAUAVPAPynwBmUEAziHkA

### 💻 Code

```ts
declare var a: void;

a ?? false;
```

### 🙁 Actual behavior

It reports "An expression of type 'void' cannot be tested for truthiness."

### 🙂 Expected behavior

The error message should not mention truthiness, since I'm not testing for truthiness.

**Comment by RyanCavanaugh** [comment]
Surely this is close enough??
